iot-hub/Samples/device/DeviceReconnectionSample/readme.md

```diff
@@ -2,6 +2,87 @@
 
 This sample code demonstrates the various connection status changes and connection status change reasons the device client can return, and how to handle them.
 
+### Initialize the client:
+
+```csharp
+// Connection string:
+// Get the device connection string from Azure IoT Portal, or using Azure CLI. 
+// Azure portal - 
+// Navigate to your IoT Hub. From the left pane, under "Explorers", click on "IoT devices".
+// Click and navigate to your device.
+// Copy the connection strings listed (primary and/or secondary).
+// Azure CLI - 
+//  az iot hub device-identity connection-string show --device-id <device_id> [--key-type {primary, secondary}]
+//  --key-type is optional. It defaults to "primary".
+//
+// Transport type:
+// The transport to use to communicate with the IoT Hub. Possible values include Mqtt,
+// Mqtt_WebSocket_Only, Mqtt_Tcp_Only, Amqp, Amqp_WebSocket_Only, Amqp_Tcp_only, and Http1.
+//
+// Pass them to the application using command line parameters (see Parameters.cs).
+
+string connectionString = "<connection_string>";
+TransportType transportType = TransportType.Mqtt;
+deviceClient = DeviceClient.CreateFromConnectionString(connectionString, transportType);
+```
+
+### Send device to cloud telemetry:
+
+```csharp
+var temperature = 25;
+var humidity = 70;
+string messagePayload = $"{{\"temperature\":{temperature},\"humidity\":{humidity}}}";
+
+using var eventMessage = new Message(Encoding.UTF8.GetBytes(messagePayload))
+{
+    MessageId = Guid.NewGuid().ToString(),
+    ContentEncoding = Encoding.UTF8.ToString(),
+    ContentType = "application/json",
+};
+
+await deviceClient.SendEventAsync(message);
+```
+
+### Receive cloud to device telemetry (using the polling API) and complete the message:
+
+```csharp
+using Message receivedMessage = await deviceClient.ReceiveAsync();
+if (receivedMessage == null)
+{
+    Console.WriteLine("No message received; timed out.");
+    return;
+}
+
+string messageData = Encoding.ASCII.GetString(receivedMessage.GetBytes());
+var formattedMessage = new StringBuilder($"Received message: [{messageData}]\n");
+foreach (var prop in receivedMessage.Properties)
```

Comment:
unclear what type `var` is

Reply:
I can update it to `KeyValuePair<string, string>`, sure